BSP/Simulator/Src/BSP_Timer.c

```diff
@@ -0,0 +1,62 @@
+#include "BSP_Timer.h"
+
+static void (*Timer1Handler)(int);
+static void (*Timer2Handler)(int);
+//had to pass some sort of parameter to the pointer or it doesn't work
+
+/** 
+* @brief   Updates the time by reading the CSV file and calling relevant functions at the right time 
+* @param   none
+* @return  none
+*/  
+void BSP_Timer_Update()
+{
+    FILE *file = fopen("Timer.csv", "r");
+    int fno = fileno(file);
+    flock(fno, LOCK_EX);
+    int timeCurrent[2];
+    int current;
+    int reload;
+    int i=0;
+    int index=0;
+    
+    while(i<2)
+    {
+        fscanf(file,"%d , %d",&current,&reload);

```

Comment:
This approach could be made cleaner if you did something like was done in BSP_CAN.c in the main simulator branch. Take a look at that and see if that would work here. It's just a bit messy with all the counters and such.

Reply:
In BSP_CAN.c, when they use the scan function, they are only passing in address of a normal int variable. My scan function also only works if I pass in only the address of an int variable.